### ISSUE TITLE
beeversion: `%Xe` and `%Xx` feature of `--format` switch cause problems

### ISSUE THREAD

**Issue by paulmenzel** [bee/bee#207]
Discussing merge request #206 with @buczek, he noted that it’s ambiguous in case you use `%ee`. And he is right.

```
$ beeversion --format "%Xe.mx64.%r\n" linux-4.8_rc2-0.bee
Xrc2.mx64.0
$ beeversion --format "%ee.mx64.%r\n" linux-4.8_rc2-0.bee
rc2erc2.mx64.0
$ beeversion --format "%eeeeee.mx64.%r\n" linux-4.8_rc2-0.bee
rc2erc2eeee.mx64.0
$ beeversion --format "%xe.mx64.%r\n" linux-4.8_rc2-0.bee
xrc2.mx64.0
$ beeversion --format "%x\n" linux-4.8_rc2-0.bee

$ beeversion --format "%x\n" linux-4.8_rc2-0-some.bee
beeversion: syntax error at position 16 in 'linux-4.8_rc2-0-some.bee'
$ beeversion --format "%x\n" linux-4.8_rc2-some-0.bee
rc2-some
$ beeversion --format "%ex\n" linux-4.8_rc2-some-0.bee
erc2-some
```

So thinking more about, I think, although a nice idea, it causes more trouble, and should be removed.

One could argue, that one could strengthen the check to make sure that only `_` and `-` are allowed right after the `%` for `e` and `x`. But I don’t like it so much either.

**Comment by mariux** [comment]
hi... 

the way it is implemented has issues when used the way you use it.. it is implemented this way because there was no reason to limit this to `_` and `-` the first place..  Too good software development is less about liking but about real usecases...

those are actually undocumented features.. ;) i am happy to merge a pull request where documentation is extended and bugs are fixed... ;) 

atm i do not see a real issue here (do you?) because that causes any real problems - or do you want to use `%ee%Xe`..

current status: it's not meant to be used with anything else but `_` and `-`.. as said in many manpages:  the results when using other chars are undefined.


```
$ beeversion --format "%Xe.mx64.%r\n" linux-4.8_rc2-0.bee
Xrc2.mx64.0
$ beeversion --format "%ee.mx64.%r\n" linux-4.8_rc2-0.bee
rc2erc2.mx64.0
$ beeversion --format "%eeeeee.mx64.%r\n" linux-4.8_rc2-0.bee
rc2erc2eeee.mx64.0
$ beeversion --format "%xe.mx64.%r\n" linux-4.8_rc2-0.bee
xrc2.mx64.0
$ beeversion --format "%ex\n" linux-4.8_rc2-some-0.bee
erc2-some
```
the results of above cases are (by new definition) undefined. (we did not care, because we don't know usecases.. pull requests to enforce this are welcome)

```
$ beeversion --format "%x\n" linux-4.8_rc2-0.bee

$ beeversion --format "%x\n" linux-4.8_rc2-0-some.bee
beeversion: syntax error at position 16 in 'linux-4.8_rc2-0-some.bee'
```
is correct/expected behavior..
```
$ beeversion --format "%x\n" linux-4.8_rc2-some-0.bee
rc2-some
```
is probably a bug... because of the `.bee` extension... i do not remember the the discussions about enforcing the revision or not in this case... @ruester do you? if not we might define it as OKish since we also could define a missing version as OK since we have the `#@beeversion` gitify feature...

**Comment by mariux** [comment]
patch available in #208 ..
